src/sidebar/components/menu.js

```diff
@@ -96,7 +106,11 @@ export default function Menu({
   // It should also close if the user presses a key which activates menu items.
   const handleMenuKeyDown = event => {
     if (event.key === 'Enter' || event.key === ' ') {
-      closeMenu();
+      // Don't close the menu right away, the link still needs to be present at least one
+      // more render cycle for them to be followed.
+      setTimeout(() => {
```

Comment:
Shy of a callback, setting the close state from its parent, and a new `open` prop, I can't remove this timeout.

Notice that closeMenu does work from useElementShouldClose, but I suspect that is become the event by then already bubbled to the body, where as in this case, it has not.